argoverse/evaluation/stereo/eval.py

```diff
@@ -0,0 +1,208 @@
+            rel_error_thresholds: Relative disparity error thresholds, in pixels.
+            save_disparity_error_image: Saves the disparity image error using the PNG format in the figs_fpath.
+            num_procs: Number of processes among which to subdivide work.
+                Specifying -1 will use one process per available core
+        """
+        self.pred_root_fpath = pred_root_fpath
+        self.gt_root_fpath = gt_root_fpath
+        self.figs_fpath = figs_fpath
+        self.abs_error_thresholds = abs_error_thresholds
+        self.rel_error_thresholds = rel_error_thresholds
+        self.save_disparity_error_image = save_disparity_error_image
+        self.num_procs = os.cpu_count() if num_procs == -1 else num_procs
+
+    def evaluate(self) -> pd.DataFrame:
+        """Evaluate stereo output and return metrics. The multiprocessing
+        library is used for parallel processing of disparity evaluation.
+
+        Returns:
+            Evaluation metrics.
+        """
+        pred_fpaths = list(self.pred_root_fpath.glob("**/*.png"))
+        gt_fpaths = list(self.gt_root_fpath.glob("**/stereo_front_left_rect_disparity/*.png"))
+        gt_obj_fpaths = list(self.gt_root_fpath.glob("**/stereo_front_left_rect_objects_disparity/*.png"))
+
+        if len(pred_fpaths) == 1:
+            timestamp = str(pred_fpaths[0]).split("_")[-1][:-4]
+            gt_fpaths = [gt_fpath for gt_fpath in gt_fpaths if timestamp in str(gt_fpath)]
+            gt_obj_fpaths = [gt_obj_fpath for gt_obj_fpath in gt_obj_fpaths if timestamp in str(gt_obj_fpath)]
+
+        assert len(pred_fpaths) == len(gt_fpaths)
+
+        if self.num_procs == 1:
+            errors = []
+            for (pred_fpath, gt_fpath, gt_obj_fpath) in zip(pred_fpaths, gt_fpaths, gt_obj_fpaths):
+                error = compute_disparity_error(
+                    pred_fpath,
+                    gt_fpath,
+                    gt_obj_fpath,
+                    self.figs_fpath,
+                    self.abs_error_thresholds,
+                    self.rel_error_thresholds,
+                    save_disparity_error_image=self.save_disparity_error_image,
+                )
+                errors.append(error)
+        else:
+            args = [
+                (
+                    pred_fpath,
+                    gt_fpath,
+                    gt_obj_fpath,
+                    self.figs_fpath,
+                    self.abs_error_thresholds,
+                    self.rel_error_thresholds,
+                    self.save_disparity_error_image,
+                )
+                for (pred_fpath, gt_fpath, gt_obj_fpath) in zip(pred_fpaths, gt_fpaths, gt_obj_fpaths)
+            ]
+            with Pool(self.num_procs) as p:
+                errors = p.starmap(compute_disparity_error, args)
+
+        data = pd.concat(errors)
+        # Sums over the instances along the rows containing the disparity error metrics.
```

Comment:
Thanks for the updates, Jhony. I believe the columns are the disparity error metrics, so it might be clearer to say
```
# Sums over all frames (row dimension) for each disparity error metric
```

Reply:
Right!